app/controllers/search_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class SearchController < ApplicationController
+  # GET /search/form
+  def form
+    @patient = Patient.new(patient_params)
+  end
+
+  # POST /search/query
+  def query
+    results = qbp_query(query_params)
+
+    case results[:code]
+    when 'AE'
+      redirect_to search_form_url,
+                  alert: 'Sorry there was an error, please try again. If problem persists contact system administrator.'
+    when 'AR'
+      render 'rejected', status: :bad_request
```

Comment:
I feel like all of these results other than `OK` could be handled by redirecting back to the search form with a flash message rather than with custom error pages.

Reply:
fixed